get-annotation.sh

```diff
@@ -0,0 +1,162 @@
+    if (( $timestamp_ms < 1328054400000 )); then
+      echo "$timestamp_ms is not a valid timestamp (looks too small, are you sure you used millis?)"
+      return 1
+    fi
+  fi
+
+  # In Bash, 0 is true, not false.
+  # See http://stackoverflow.com/questions/5431909/bash-functions-return-boolean-to-be-used-in-if.
+  return 0
+}
+
+while getopts "s:a:b:e:n:d:i:h" OPTION
+do
+  case $OPTION in
+    h)
+      usage
+      exit 1
+      ;;
+    s)
+      HOST=$OPTARG
+      ;;
+    a)
+      API_TOKEN=$OPTARG
+      ;;
+    b)
+      DATE_START=$OPTARG
+      ;;
+    e)
+      DATE_END=$OPTARG
+      ;;
+    n)
+      TENANT_NAME=$OPTARG
+      ;;
+    d)
+      DOMAIN_ID=$OPTARG
+      ;;
+    i)
+      ANNOTATION_ID=$OPTARG
+      ;;
+    ?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+
+which jq > /dev/null
+if [ $? -ne 0 ]; then
+  echo "The jq utility must be installed to run this script."
+  echo "See https://stedolan.github.io/jq/"
+  exit 1
+fi
+
+if [ -z "$API_TOKEN" ]; then
+  echo "API token is required (use the -a parameter)."
+  exit 1
+fi
+
+if [ -z "$ANNOTATION_ID" ] && [ -z "$DATE_START" ]; then
+  echo "You must supply either an Annotation ID (-i) or a Start Date (-b)."
+  exit 1
+else
+  if [ -n "$DATE_START" ]; then
+    if [ -z "$TENANT_NAME" ]; then
+      echo "Tenant Name is required (use the -n parameter)."
+      exit 1
+    fi
+    # Make sure the timestamp is valid.
+    if ! validTimestamp "$DATE_START"; then
+      echo "date-start is invalid.  It must be in epoch millis."
+      exit 1
+    else
+      if [ -n "$DATE_END" ]; then
+        # Make sure the timestamp is valid.
+        if ! validTimestamp "$DATE_END"; then
+          echo "date-end is invalid.  It must be in epoch millis."
+          exit 1
+        fi
+      else
+        # No timestamp provided.
+        # Get the current time in epoch millis.
+        DATE_END=$(date +%s)000
+      fi
+      if [ -n "$DOMAIN_ID" ]; then
+        URL="https://$HOST/concerto/mpulse/api/annotations/v1?date-start=$DATE_START&date-end=$DATE_END&domain=$DOMAIN_ID"
+      else
+        URL="https://$HOST/concerto/mpulse/api/annotations/v1?date-start=$DATE_START&date-end=$DATE_END"
+      fi
+    fi
+  else
+    URL="https://$HOST/concerto/mpulse/api/annotations/v1/$ANNOTATION_ID"
+  fi
+fi
+
+
+credentials_json_file=$(mktemp)
+cat > $credentials_json_file << EOF
+{ "apiToken": "$API_TOKEN", "tenant": "$TENANT_NAME" }
+EOF
+
+token_json=$(curl --fail --silent --show-error -X PUT --data-binary @$credentials_json_file https://$HOST/concerto/services/rest/RepositoryService/v1/Tokens)
+curl_result=$?
+rm $credentials_json_file
+
+if [ $curl_result -ne 0 ]; then
+  echo "Failed to obtain a security token (curl exit code: $curl_result)."
+  exit 1
+fi
+
+token_quoted=$(echo $token_json | jq .token)
+token_raw=$(echo $token_quoted | sed 's/\"//g')
+
+echo $URL
+curl --fail --silent --show-error -H "X-Auth-Token: $token_raw" -X GET $URL | jq
+curl_result=$?
```

Comment:
@wsmithee this won't work; instead of getting the exit code for `curl`, you'll get the exit code for `jq`.  You'll need to instead capture the `curl` output to a variable, then check the exit code, then echo it to `jq` if curl succeeded.